### DISCUSSION TITLE
How to dynamically toggle a packageRule without changing the config.js?

### DISCUSSION THREAD

**Discussion by phaninra** [renovatebot/renovate#18008] [Request Help] [answered]
Sample setup

`packageRules: [
     {
     // rule 1
     },
     {
     // rule 2
     }
]`

 I'd like to control the application of rule1 or rule2 or both at runtime. So far I've found 1 way to do it by utilizing `schedule`

```
const rule1Schedule = process.env.RULE1_SCHEDULE // set this to "at any time"
const rule2Schedule = process.env.RULE2_SCHEDULE  // set this to "before 12am" i.e. never

module.exports = {
    packageRules: [
           {
             // rule 1
            schedule: rule1Schedule 
           },
           {
           // rule 2
            schedule: rule2Schedule 
           }
   ]
}
```

Above setup will ensure that only rule1 is applied. 

Just wondering if there's another way to do this. In addition, the `before 12am` worked so far in my experiments but I'm not so sure about that. Open to inputs.

**Comment by rarkins** [comment, marked as answer]
It's a JS file so you can do pretty much anything you want. Including awaiting async calls:

```
module.exports = async function () {
  const projectsResult = await got(endpoint + '/projects', { ... });
  ...
  return { ... }
}
```
